src/main/groovy/com/netgrif/application/engine/startup/DefaultFiltersRunner.groovy

```diff
@@ -242,9 +249,16 @@ class DefaultFiltersRunner extends AbstractOrderedCommandLineRunner {
             return Optional.empty()
         }
 
-        def systemUser = this.userService.getLoggedOrSystem()
+        def loggedUser = this.userService.getLoggedOrSystem()
+
+        if (loggedUser.getStringId() == this.userService.getSystem().getStringId()) {
+            Case filterCase = this.workflowService.searchOne(QCase.case$.processIdentifier.eq("filter").and(QCase.case$.title.eq(title)).and(QCase.case$.author.id.eq(userService.getSystem().getStringId())))
+            if (filterCase != null) {
+                return Optional.of(filterCase)
+            }
+        }
 
-        Case filterCase = this.workflowService.createCase(filterNet.getStringId(), title, null, systemUser.transformToLoggedUser()).getCase()
+        Case filterCase = this.workflowService.createCase(filterNet.getStringId(), title, null, loggedUser.transformToLoggedUser()).getCase()
```

Comment:
is the `loggedUser` always set? Is it possible for this method to be called with `loggedUser` being `null`?

Reply:
def loggedUser = this.userService.getLoggedOrSystem()